kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ConfigurationDefinition.java

```diff
@@ -12,5 +12,6 @@
  * Defines the details of how Contributions interact with configuration for {@link BaseContributor}'s purposes.
  *
  * @param configurationType defines the expected class for configuration objects
+ * @param configurationRequired {@code true} if the contribution requires configuration. {@code false} if the contribution uses configuration for optional properties or to override defaults.
```

Comment:
Wondering if it's right to document the purpose of setting it false like this. 

Shouldn't it talk more about the nullness of configuration. Like `{@code true} if the contribution requires a non-null configuration object. {@code false} if the contribution can tolerate a null configuration object.` Then maybe add the purpose bit in another sentence or brackets.

Reply:
non null is certainly what it currently means, but can we offer a more useful definition? 

I really want config required to be true if there are essential properties to be supplied.